core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -503,6 +503,23 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
         set("spark.executor.instances", value)
       }
     }
+
+    if (contains("spark.master") && get("spark.master").startsWith("yarn")) {
```

Comment:
Don't you need to check `startsWith("yarn-")`? because "yarn" starts with "yarn".